test-suite/src/test/scala/org/scalajs/testsuite/javalib/ListTest.scala

```diff
@@ -0,0 +1,270 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib
+
+import scala.language.implicitConversions
+
+import org.scalajs.jasminetest.JasmineTest
+
+trait ListTest extends JasmineTest {
+
+  val listFactory: ListFactory
+
+  trait ListFactory {
+    def empty[E]: java.util.List[E]
+  }
+
+  def expectIndexOutOfBoundsException[T](f: => T): Unit = {
+    expect({
+      try {
+        f
+        false
+      } catch {
+        case err: IndexOutOfBoundsException => true
+      }
+    }).toBeTruthy
+  }
+
+  def testListApi() = {
+    it("should store strings") {
+      val al = listFactory.empty[String]
+
+      expect(al.size()).toEqual(0)
+      al.add("one")
+      expect(al.size()).toEqual(1)
+      expect(al.get(0)).toEqual("one")
+      al.add("two")
+      expect(al.size()).toEqual(2)
+      expect(al.get(0)).toEqual("one")
+      expect(al.get(1)).toEqual("two")
+
+      expectIndexOutOfBoundsException(al.get(-1))
+      expectIndexOutOfBoundsException(al.get(al.size))
+    }
+
+    it("should store integers") {
+      val al = listFactory.empty[Int]
+
+      al.add(1)
+      expect(al.size()).toEqual(1)
+      expect(al.get(0)).toEqual(1)
+      al.add(2)
+      expect(al.size()).toEqual(2)
+      expect(al.get(0)).toEqual(1)
+      expect(al.get(1)).toEqual(2)
+
+      expectIndexOutOfBoundsException(al.get(-1))
+      expectIndexOutOfBoundsException(al.get(al.size))
+    }
+
+    it("should store doubles also in corner cases") {
```

Comment:
These are not really corner cases. You'll get actual corner cases when using `indexOf` and `lastIndexOf`. For example, looking for `+0.0` should not find `-0.0` even if the latter comes before. And it should be possible find `NaN`.